### PR TITLE
feat: Keymaps config file.

### PR DESCRIPTION
**Pull request by Zaa3ter**
- Externalized Configuration: Added support for loading keybindings from a JSON configuration file located in the application's configuration directory (SurgeDir).
- Modularization: Refactored the codebase to move hardcoded keybinding definitions out of internal/tui/keys.go and into a new, dedicated configuration management system in internal/config/keymaps.go as same style as internal/config/keymaps.go.
- Configurable Keymaps: Implemented the KeyMap structure and loading logic to allow users to remap bindings for various application components, including the Dashboard, Input, File Picker, and other interactive elements.
- Verification: Included comprehensive unit tests (internal/config/keymaps_test.go) to ensure correct loading and parsing of the keymap configurations.

<h3>Greptile Summary</h3>

This PR externalizes keybindings from a hardcoded `internal/tui/keys.go` into a new `internal/config/keymaps.go` that loads from and writes to a `keymap.json` file in the application config directory, with reflection-based apply/validate helpers and startup warnings surfaced in the activity log.

- `internal/config/keymaps.go`: introduces `LoadKeyMap`/`SaveKeyMap`, a reflection-driven `ApplyConfig`, a `Validate` helper, and a full `DefaultKeyMap` — migrated verbatim from the old TUI package, wrapped in JSON serialization.
- `internal/tui/keys.go` is reduced to a one-liner (`var Keys = config.DefaultKeyMap()`), and `RootModel.keys` becomes `*config.KeyMap` loaded at startup; a nil guard is added at the top of `Update()` for zero-value test models.
- `internal/config/keymaps_test.go` covers round-trip conversion and the `Validate` helper, but `LoadKeyMap` itself cannot be called from tests because `GetSurgeDir()` has no injection point.

<h3>Confidence Score: 5/5</h3>

Safe to merge; all new code paths fall back to defaults on any failure and the happy path is well-exercised by existing tests.

The change is a clean extraction of already-working keybinding definitions into a config-backed layer. Load/save paths degrade gracefully to defaults, startup warnings surface failures non-fatally, and the TUI update loop has an explicit nil guard. Remaining gaps are quality items that do not affect runtime correctness.

internal/config/keymaps_test.go — the test suite bypasses LoadKeyMap entirely due to the hardcoded GetSurgeDir(), leaving the file-not-found and JSON-corruption branches without direct coverage.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| internal/config/keymaps.go | New keymap config file with load/save/apply/validate logic; contains one dead guard condition in Validate() and pre-existing issues already covered in earlier review threads |
| internal/config/keymaps_test.go | Tests cover DefaultKeyMap, ToConfig/ApplyConfig round-trip, and Validate, but LoadKeyMap itself is never called because GetSurgeDir() is not injectable, leaving all three branches of the loader untested |
| internal/tui/keys.go | Gutted to a one-liner re-export; the global Keys var changed from a copied struct to a shared pointer, creating latent test-isolation risk |
| internal/tui/model.go | keys field changed from value to pointer type; LoadKeyMap error is discarded and nil-guarded correctly |
| internal/tui/update.go | Adds a simple nil guard for m.keys at the top of Update(); straightforward and correct |
| internal/tui/update_test.go | Minor test fix to add inputs to the nil-settings test model; no issues |

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 3 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 3
internal/config/keymaps.go:340
**Dead type guard in `Validate`**

The condition `fieldType != reflect.TypeFor[[]string]()` is unreachable. `[]string` has `Kind() == reflect.Slice`, so it can never satisfy the preceding `fieldType.Kind() == reflect.Struct` check. The inner guard is always `true` when the outer one is `true`, and can be dropped without changing behaviour.

### Issue 2 of 3
internal/config/keymaps_test.go:860-862
**`LoadKeyMap` itself is never exercised by the test suite**

The comment acknowledges that `LoadKeyMap` cannot be called because `GetSurgeDir()` is a package-level function with no injection point. As a result, the file-not-found path (creates defaults), the JSON-corruption path (overwrites and warns), and the normal parse path are all untested through the public API. Accepting a path parameter or injecting a dir-resolver function into `LoadKeyMap` would make it directly testable and bring coverage of those three branches.

### Issue 3 of 3
internal/tui/keys.go:6
**`Keys` is now a shared mutable pointer**

The old `var Keys = KeyMap{...}` was a value; every `keys: Keys` assignment in a test copied the whole struct. Now `Keys` is `*config.KeyMap`, so every `keys: Keys` (found in `update_test.go`, `delete_resilience_test.go`, `category_regressions_test.go`, `settings_restart_test.go`, and `keys_test.go`) shares the same underlying object. No current code path writes through `m.keys` after init, so tests pass today, but any future code that mutates a field via the pointer would silently corrupt every other test using the same pointer. A `func DefaultKeys() *config.KeyMap { return config.DefaultKeyMap() }` helper would make isolation explicit.

`````

</details>

<sub>Reviews (5): Last reviewed commit: ["fix: recreate the keymap file when there..."](https://github.com/surgedm/surge/commit/e71bc60ca06adbec3740f693e85071c3da8e7f10) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32203231)</sub>

**Context used:**

- Rule used - What: All code changes must include tests for edge... ([source](https://app.greptile.com/review/custom-context?memory=2b22782d-3452-4d55-b059-e631b2540ce8))

### CODE REVIEW

File: internal/tui/model.go [hunk 1]
```diff
@@ -292,11 +292,19 @@ func InitialRootModel(serverPort int, currentVersion string, service core.Downlo
 		settings = config.DefaultSettings()
 	}
 
+	keys, _ := config.LoadKeyMap()
+	if keys == nil {
+		keys = config.DefaultKeyMap()
+	}
```
[the remark this answers is not in the document]
Comment (Zaa3ter) [reply internal/tui/model.go:298]: There are no provided pattern in the existed code to deal with how to respond to IO Error from opening a file for like in `func LoadSettings() (*Settings, error)`

**Comment by junaid2005p** [comment]
Hey @Zaa3ter thanks for contributing I will take a look at this as soon as I can

**Comment by SuperCoolPencil** [comment]
Hi @Zaa3ter! Thank you so much for this amazing contribution. We love both of your PRs (#443 and #449)

I have merged the changes from this PR (#449) directly into your Vim keybindings PR (#443) so they can be shipped together under a unified configuration!

Since all the configuration-backed keymap changes from here are now fully integrated and all tests/linters are clean over on #443, I will close this PR. We'll be merging the combined work via #443 shortly.

Thank you again for your wonderful contributions to Surge! 🚀

**Comment by Zaa3ter** [comment]
You’re welcome, and thank you for the opportunity.
